commands/utils/extractors.go

```diff
@@ -0,0 +1,87 @@
+package utils
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/jfrog/build-info-go/build"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+)
+
+var extractorsRepositoryPath = filepath.Join("artifactory", "oss-release-local")
+
+type extractorDetails struct {
+	extractorType string
+	localPath     string
+	remotePath    string
+	fileName      string
+}
+
+func (ed *extractorDetails) DownloadToPath() string {
+	return filepath.Join(ed.localPath, ed.fileName)
+}
+
+func (ed *extractorDetails) DownloadFromPath() string {
+	return filepath.Join(extractorsRepositoryPath, ed.remotePath, ed.fileName)
+}
+
+// downloadExtractorsFromRemoteIfNeeded downloads build-info-extractors for air-gapped environments, if they're not exist on the remote repository yet.
+func downloadExtractorsFromRemoteIfNeeded(server *config.ServerDetails, remoteRepoName string, extractorsLocalPath string) (err error) {
+	log.Info("Downloading extractors if needed from", remoteRepoName)
```

Comment:
Let's replace this log message with "Checking whether the build-info extractors exist locally..."